src/intellij2024/kotlin/com/sourcegraph/cody/autocomplete/CodyInlineCompletionProvider.kt

```diff
@@ -0,0 +1,147 @@
+package com.sourcegraph.cody.autocomplete
+
+import com.intellij.codeInsight.inline.completion.*
+import com.intellij.codeInsight.inline.completion.elements.InlineCompletionGrayTextElement
+import com.intellij.codeInsight.inline.completion.suggestion.InlineCompletionSingleSuggestion
+import com.intellij.codeInsight.inline.completion.suggestion.InlineCompletionSuggestion
+import com.intellij.openapi.application.ApplicationInfo
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ReadAction
+import com.intellij.openapi.client.ClientSessionsManager
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.event.DocumentEvent
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.util.concurrency.annotations.RequiresReadLock
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.AutocompleteResult
+import com.sourcegraph.cody.agent.protocol.CompletionItemParams
+import com.sourcegraph.cody.statusbar.CodyStatusService.Companion.resetApplication
+import com.sourcegraph.cody.vscode.CancellationToken
+import com.sourcegraph.cody.vscode.InlineCompletionTriggerKind
+import com.sourcegraph.cody.vscode.IntelliJTextDocument
+import com.sourcegraph.config.ConfigUtil
+import com.sourcegraph.utils.CodyEditorUtil.getTextRange
+import com.sourcegraph.utils.CodyEditorUtil.isImplicitAutocompleteEnabledForEditor
+import com.sourcegraph.utils.CodyFormatter
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+
+class CodyInlineCompletionProvider : InlineCompletionProvider {
+  private val logger = Logger.getInstance(CodyInlineCompletionProvider::class.java)
+  private val currentJob = AtomicReference(CancellationToken())
+  override val id = InlineCompletionProviderID("Cody")
+
+  override suspend fun getSuggestion(request: InlineCompletionRequest): InlineCompletionSuggestion {
+    ApplicationManager.getApplication().assertIsNonDispatchThread()
+    val editor = request.editor
+    val project = editor.project ?: return InlineCompletionSuggestion.Empty
+    if (!isImplicitAutocompleteEnabledForEditor(editor)) {
+      return InlineCompletionSuggestion.Empty
+    }
+    val lookupString: String? = null // todo: can we use this provider for lookups?
```

Comment:
Can you elaborate with `// TODO: what, when` so it is clear what is actionable when here?

Reply:
This is on me. I left it when I was implementing inline completion provider for the remote. I will revisit this item.

Reply:
https://linear.app/sourcegraph/issue/CODY-4137/add-support-for-lookup-strings-for-the-remotes-autocompletions